tests/test-definitions.sh

```diff
@@ -0,0 +1,254 @@
+    
+    # Verify the compilation of CUDA samples
+    /usr/local/cuda/samples/0_Introduction/mergeSort/mergeSort
+    check_exit_code "CUDA Samples $cuda" "Failed to perform merge sort using CUDA Samples"
+}
+
+function verify_nccl_installation {
+    # Print nccl.conf if it exists
+    if test -f /etc/nccl.conf; then
+        cat /etc/nccl.conf
+    fi
+
+    module load mpi/hpcx
+
+    # Check the type of Mellanox card in use
+    # mellanox_card=$(lspci -nn | grep -m 1 Mellanox | awk '{print $9 " " $10}' | sed 's/\[//' | sed 's/\]//')
+    
+    # Run NCCL test based on Mellanox card
+    case $VMSIZE in
+        standard_nc24rs_v3) mpirun -np 4 \
+            -x LD_LIBRARY_PATH \
+            --allow-run-as-root \
+            --map-by ppr:4:node \
+            -mca coll_hcoll_enable 0 \
+            -x UCX_TLS=tcp \
+            -x CUDA_DEVICE_ORDER=PCI_BUS_ID \
+            -x NCCL_SOCKET_IFNAME=eth0 \
+            -x NCCL_DEBUG=WARN \
+            /opt/nccl-tests/build/all_reduce_perf -b1K -f2 -g1 -e 4G;;
+        standard_nd40rs_v2 | standard_nd96*v4 | standard_nc*ads_a100_v4) mpirun -np 8 \
+            --allow-run-as-root \
+            --map-by ppr:8:node \
+            -x LD_LIBRARY_PATH=/usr/local/nccl-rdma-sharp-plugins/lib:$LD_LIBRARY_PATH \
+            -mca coll_hcoll_enable 0 \
+            -x UCX_TLS=tcp \
+            -x CUDA_DEVICE_ORDER=PCI_BUS_ID \
+            -x NCCL_SOCKET_IFNAME=eth0 \
+            -x NCCL_DEBUG=WARN \
+            -x NCCL_NET_GDR_LEVEL=5 \
+            /opt/nccl-tests/build/all_reduce_perf -b1K -f2 -g1 -e 4G;;
+        *) ;;
+    esac
+    check_exit_code "NCCL $nccl" "Failed to run NCCL all reduce perf"
+    
+    module unload mpi/hpcx
+}
+
+function verify_spack_installation {
+    spack --version
+    check_exit_code "Spack $spack" "Failed to install Spack"
+}
+
+function verify_azcopy_installation {
+    sudo azcopy --version
+    check_exit_code "azcopy $azcopy" "Failed to install azcopy"
+}
+
+function verify_mkl_installation {
+    local intelmkl_path=$(spack location -i intel-oneapi-mkl@$intel_one_mkl)
+    check_exists $intelmkl_path
+    check_exit_code "Intel Oneapi MKL $intel_one_mkl" "Intel Oneapi MKL installation not found!"
+}
+
+function verify_hpcdiag_installation {
+    local hpcdiag_path="$HPC_ENV/diagnostics/gather_azhpc_vm_diagnostics.sh"
+    check_exists $hpcdiag_path
+}
+
+# Internal/ external installation of GCC
+function verify_gcc_installation {
+    gcc --version
+    check_exit_code "GCC is installed" "GCC doesn't exist!"
+}
+
+# Check module file for the explicit installations
+function verify_gcc_modulefile {
+    # Verify GCC Software installation path
+    check_exists "/opt/gcc-$gcc_version/"
+    # Verify GCC module file path
+    check_exists "$MODULE_FILES_ROOT/gcc-$gcc_version"
+}
+
+function verify_aocl_installation {
+    # verify AMD modulefiles
+    check_exists "$MODULE_FILES_ROOT/amd/aocl"
+    check_exists "/opt/amd/lib/"
```

Comment:
Is this how AMD recommends installing? Directly into /opt/amd/, not into a named subdir of /opt/amd/ ?

Reply:
We usually install it in the /opt/amd dir